tmplctlr/kubeclient.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2017 the lostromos Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tmplctlr
+
+import (
+	"os"
+	"os/exec"
+)
+
+// KubeClient is an interface that implements an Apply() and Delete() for our K8s templates
+type KubeClient interface {
+	Apply(file string) (string, error)
+	Delete(file string) (string, error)
+}
+
+// Kubectl provides a simple wrapper around calling the needed kubectl commands
+// TODO: This should be revisited when https://github.com/kubernetes/kubernetes/issues/15894 is completed.
+// #15894 will move the apply logic from kubectl into the API
+type Kubectl struct {
+	ConfigFile string //config file for kubectl
+}
+
+var execCommand = exec.Command
+
+// Apply will execute kubectl apply -f file with the correct config
```

Comment:
Surely there's a way we can avoid shelling out to do this. Is any of the K8S code such that we could call in directly to do the apply? (from what I know of the API, we can't do that directly).

Reply:
Currently we can't. The code that get's called for apply is 200+ lines long and requires all of the kubectl command and variables and everything to be present. :(